mountpoint-s3-client/src/object_client.rs

```diff
@@ -32,6 +32,7 @@ pub trait ObjectClient {
         bucket: &str,
         key: &str,
         range: Option<Range<u64>>,
+        if_match: Option<String>,
```

Comment:
When should we switch to using a request struct rather than adding more objects?

Maybe now is a good time, with these two optional arguments. We'll want to start adding more arguments for [additional checksums](https://aws.amazon.com/blogs/aws/new-additional-checksum-algorithms-for-amazon-s3/) in the near future too.

Reply:
What is meant by request struct here?

Reply:
I think 4 arguments is probably right on the limit for this, so I'd leave it as is for this PR, and we can do it the next time we find ourselves wanting to add an argument. Maybe just add a `// TODO convert to a builder pattern if we add more arguments` comment.

Reply:
If we leave bucket and key, following are the only 2 arguments for request struct.
`pub struct request {
     range: Option<Range<u64>>,
     if_match: Option<String>,
}`